### DISCUSSION TITLE
Infer Types

### DISCUSSION THREAD

**Discussion by IrfanAli17899** [mrjono1/joi-to-typescript#362] [Ideas]
Hi, this is very useful tool, thanks for this :raised_hands: 
i want to confirm that is there any feature in it that allows to simply infer types instead of generating types files, jut like yup or zod?

```
type SchemaType = yup.InferType<typeof schema>;
type SchemaType = z.infer<typeof schema>;
```

**Comment by mrjono1** [comment]
This package doesn't infer types, I did see something a while ago for extending joi to add them.
If you can figure out how to add infer types that would be interesting
